tests/test_pip.py

```diff
@@ -191,3 +191,10 @@ def test_getPackages_error(
 ERROR: Could not find a version that satisfies the requirement {packageName} (from versions: none)
 ERROR: No matching distribution found for {packageName}""".lower()
     )
+
+
+def test__convertPipReport(data_root_dir: str):
+    # this file contains special characters that might lead to UnicodeDecodeError
+    reportRezPipPath = os.path.join(data_root_dir, "pip", "pip-install.report.rez-pip")
```

Comment:
I think I would prefer to see a more targeted test. Instead of a `readPipReport` function, I would just add a `readReport` that would return a JSON. There is two reasons for that. First, converting the report is already handled in other tests and second, I would prefer to have the data in the test itself instead of in a separate file and we don't really need to have real world data to test characters encoding. The data could be a really simple raw JSON string with exotic (non latin) characters.

Reply:
make sense !